ui/src/main/java/org/pentaho/di/ui/spoon/trans/LogBrowser.java

```diff
@@ -122,51 +123,54 @@ public void run() {
+                lastNr = Math.min( lastNr, lastLogId.get() + MAX_NR_LOG_LINES_CHUNK );
+
                 List<KettleLoggingEvent> logLines =
                   KettleLogStore.getLogBufferFromTo( childIds, true, lastLogId.get(), lastNr );
 
-                // The maximum size of the log buffer
-                //
-                int maxSize = Props.getInstance().getMaxNrLinesInLog() * 150;
-
-                // int position = text.getSelection().x;
-                // StringBuilder buffer = new StringBuilder(text.getText());
-
                 synchronized ( text ) {
 
-                  for ( int i = 0; i < logLines.size(); i++ ) {
-                    KettleLoggingEvent event = logLines.get( i );
+                  for ( KettleLoggingEvent event : logLines ) {
                     String line = logLayout.format( event ).trim();
                     int start = text.getText().length();
                     int length = line.length();
 
                     if ( length > 0 ) {
                       Format format = TextFormatter.getInstance().execute( line );
-                      text.append( format.getText() );
+                      String theRealText = format.getText();
+                      // The formatting may have change the original text and its length
+                      length = theRealText.length();
+                      text.append( theRealText );
                       text.append( Const.CR );
 
                       for ( StyleRange styleRange : format.getStyleRanges() ) {
                         styleRange.start += start;
                         text.setStyleRange( styleRange );
                       }
 
+                      StyleRange styleRange = new StyleRange();
+                      styleRange.start = start;
+
                       if ( event.getLevel() == LogLevel.ERROR ) {
-                        StyleRange styleRange = new StyleRange();
-                        styleRange.foreground = GUIResource.getInstance().getColorRed();
-                        styleRange.start = start;
+                        styleRange.foreground = colorRed;
                         styleRange.length = length;
-                        text.setStyleRange( styleRange );
                       } else {
-                        StyleRange styleRange = new StyleRange();
-                        styleRange.foreground = GUIResource.getInstance().getColorBlue();
-                        styleRange.start = start;
+                        styleRange.foreground = colorBlue;
                         styleRange.length = Math.min( 20, length );
-                        text.setStyleRange( styleRange );
                       }
+
+                      text.setStyleRange( styleRange );
                     }
                   }
                 }
 
+                // The maximum size of the log buffer
+                //
+                int maxSize = Props.getInstance().getMaxNrLinesInLog() * 150;
```

Comment:
Can the 150 be replaced by the newly created variable MAX_NR_LOG_LINES_CHUNK ?

Reply:
It's a coincidence the '150' value as, here, it's referring to the estimated size of a log line.
I'll extract it to a new constant with a self-explaining name.